src/factory.js

```diff
@@ -193,6 +194,15 @@ module.exports = ({ Component, createElement, PropTypes }) => {
           ))
           const completeStateKeys = (this._stateKeys = keys(stateDescriptors))
 
+          this._resetState = () => {
+            const _initialstate = initialState(props)
```

Comment:
IMHO we should use current props here (`this.props`) instead of initial props (`props` param).